### PR TITLE
merge definiitions with relocatable tables to ease understanding and maintenance

### PR DESCRIPTION
**Pull request by patricksurry**
the extra asm pass is a good idea.  This has no functional change, it just merges the definition and declaration of the relocatable zero page and user page tables which makes them much easier to understand and keep in sync.  

The tables end up at the end of native words instead of just after cold but that doesn't change any behavior.

Let me know if you're OK with this approach before I make any changes to where variables live etc.

### CODE REVIEW

File: definitions.asm [hunk 1]
```diff
@@ -7,103 +7,140 @@
...
+buffstatus_offset:      .word 0         ; Buffer status (bit 0 = used, bit 1 = dirty) (not in use)
 
 ; Block I/O vectors
-blockread_offset    = 46   ; Vector to block reading routine
-blockwrite_offset   = 48   ; Vector to block writing routine
+
+blockread_offset:       .word xt_block_word_error   ; Vector to block reading routine
+blockwrite_offset:      .word xt_block_word_error   ; Vector to block writing routine
+
+                        .word 0          ; 'COLD
```
Comment (patricksurry) [inline definitions.asm]: Do you know what this is for?  I couldn't find a reference to it in the code.

File: definitions.asm [hunk 2]
```diff
@@ -7,103 +7,140 @@
...
+; overflow floodplain.
+
+; This table defines all of the zero page variables along with their initial
+; values except the uninitialized temporaries at the end.  This table
+; is relocated to zero page by COLD.
+
+dsp0      = zpage_end-7    ; initial Data Stack Pointer
+
+cold_zp_table:
+        .logical user0
```
Comment (patricksurry) [inline definitions.asm]: This generates labels for the relocated destination.

File: definitions.asm [hunk 3]
```diff
@@ -7,103 +7,140 @@
...
+        ;       = 0 CTRL-n recalls previous history
+        ; Bit 2 = Current history buffer msb
+        ; Bit 1 = Current history buffer (0-7, wraps)
+        ; Bit 0 = Current history buffer lsb
+        ;
+        ; status+1 is used by ACCEPT to hold history lengths.
+
+; The remaining ZP variables are uninitialized temporaries.
+
+    .virtual
```
Comment (patricksurry) [inline definitions.asm:82]: This generates correct labels but emits no data.

File: definitions.asm (hunk 1 above)
Comment (SamCoVT) [reply definitions.asm]: I do not know where this came from.  BLAME shows you added it 3 weeks ago in [prune some code, add comments](https://github.com/SamCoVT/TaliForth2/commit/9db6ed4455663ff9e93b1c830814d2b774e21162)

**Comment by SamCoVT** [comment]
I think this approach looks good.  64tass is way more powerful than Ophis (the assembler we were using previously), but we've really only used the conditional assembling up to this point.  The only sad note is that this will be one more hurdle for people trying to port to a different assembler, but they should be able to recreate the labels without _too_ much hassle and this will make less of a chance of oopsies by having the labels and starting values in two different places.

Let me know when you like this and I will merge it.

**Comment by patricksurry** [comment]
haha, that's funny I have no recollection of where it came from.

I've mostly used `ca65` in the past which does a decent job with segments but needs a separate linker config file.

agree it'd be good to minimize use of special assembler features but this does seem worth it here.

I'm good with this as a place to build off.

**Comment by SamCoVT** [comment]
One note about the `.logical` and `.virtual` assembler directives (post merging):  
This is apparently a new feature of 64tass - I had to upgrade for it to assemble with these new directives.  I was on 1.54.1900 (did not work - I installed it in 2020) and am now on 1.56.2625 (assembles fine, this version released in 2021) by asking my package manager to install it for me (it's not the newest version, but is new enough).  It might be worth adding 1.56.2625 as the minimum version for 64tass in the docs (I'll add it to my list of "not critical, but probably should do" updates for the docs).

**Comment by patricksurry** [comment]
Interesting. I’m new to this assembler so wasn’t thinking about recency.
Iirc there’s a note in the doc that one of logical/virtual was renamed from
some confusing legacy name which is now deprecated

**Pull request merged**